roboteam_ai/src/control/controlGoToPos/ControlGoToPosLuTh.cpp

```diff
@@ -17,37 +16,78 @@ void ControlGoToPosLuTh::clear() {
 Vector2 ControlGoToPosLuTh::goToPos(RobotPtr robot, Vector2 &target) {
     Vector2 velocityCommand;
 
+    if (! pidInit) {
+        pidInit = true;
+        pid.setPID(constants::standard_luth_P,
+                constants::standard_luth_P,
+                constants::standard_luth_P);
+    }
+
+    if (pid.getP() != interface::InterfaceValues::getLuthP() ||
+            pid.getI() != interface::InterfaceValues::getLuthI() ||
+            pid.getD() != interface::InterfaceValues::getLuthD()) {
+
+        pid.reset();
+        pid.setPID(interface::InterfaceValues::getLuthP(),
+                interface::InterfaceValues::getLuthI(),
+                interface::InterfaceValues::getLuthD());
+
+    }
+
+    useRobotIndex = false;
     bool recalculate = false;
     double deltaTarget = (abs((target - targetPos).length()));
     double deltaPos = (abs((target - robot->pos).length()));
 
-    if (deltaTarget > errorMargin && ! (deltaPos < errorMargin*4.0 && deltaTarget > errorMargin*2.0)) {
+    if (deltaTarget > errorMargin && ! (
+            deltaPos < errorMargin*4.0 && deltaTarget > errorMargin
+                    *2.0)) {
         recalculate = true;
     }
-    else if (me.posData.size() > 4) {
+    else if (me.posData.
+            size()
+            > 4) {
 
         auto robotPos = static_cast<Vector2>(robot->pos);
         int currentIndex = 0;
         double distance = 999999;
-        for (int i = 0; i < static_cast<int>(me.posData.size()); i ++) {
-            me.pos = me.posData[i];
-            me.t = me.posData.size()*me.dt;
+        for (
+                int i = 0;
+                i < static_cast
+                        <int>(me
+                                .posData.
+                                size()
+                ); i ++) {
+            me.
+                    pos = me.posData[i];
+            me.
```

Comment:
dude

Reply:
yes wtf :0 probably ctrl alt L going wrong or something
fixed it